flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/utils/AvroKryoSerializerUtils.java

```diff
@@ -106,11 +105,55 @@ public void write(Kryo kryo, Output output, Schema object) {
         }
 
         @Override
-        public Schema read(Kryo kryo, Input input, Class<Schema> type) {
+        public Schema read(Kryo kryo, Input input, Class<? extends Schema> type) {
             String schemaAsString = input.readString();
             // the parser seems to be stateful, to we need a new one for every type.
             Schema.Parser sParser = new Schema.Parser();
             return sParser.parse(schemaAsString);
         }
     }
+
+    public static class AvroGenericDataArraySerializer
+            extends CollectionSerializer<GenericData.Array> implements Serializable {
+        @Override
+        public void write(Kryo kryo, Output output, GenericData.Array collection) {
+            String schemaAsString = collection.getSchema().toString(false);
```

Comment:
Can the collection be null here?

Reply:
I do notice that the Kryo library built-in serializers check for null, but I've added unit tests for the null scenario and I'm testing both the Flink serializer and the Kryo serializer directly and I can't create a scenario where null gets sent to that method.